frontend/src/views/SecretOverviewPage/SecretOverviewPage.tsx

```diff
@@ -173,11 +173,31 @@ export const SecretOverviewPage = () => {
   }, [isWorkspaceLoading, workspaceId, router.isReady]);
 
   const userAvailableEnvs = currentWorkspace?.environments || [];
-  const [visibleEnvs, setVisibleEnvs] = useState(userAvailableEnvs);
+  const [visibleEnvs, setVisibleEnvs] = useState(
+    userAvailableEnvs?.filter(({ slug }) =>
+      permission.can(
+        ProjectPermissionActions.Read,
+        subject(ProjectPermissionSub.Secrets, {
+          environment: slug,
+          secretPath
+        })
+      )
+    )
+  );
 
   useEffect(() => {
-    setVisibleEnvs(userAvailableEnvs);
-  }, [userAvailableEnvs]);
+    setVisibleEnvs(
+      userAvailableEnvs?.filter(({ slug }) =>
```

Comment:
Can we also hide it here?

![CleanShot 2024-09-14 at 23 21 12@2x](https://github.com/user-attachments/assets/69cbd311-ce69-4f47-8d24-8ed453e0f17f)

Reply:
Nice catch!
<img width="522" alt="image" src="https://github.com/user-attachments/assets/ebbb17b9-c241-430a-8a78-f6fb42648a56">

I noticed I missed it here as well, fixed:
<img width="649" alt="image" src="https://github.com/user-attachments/assets/b2feb936-ae61-42ad-8fca-337d4deea45b">